util/data_set_helpers.py

```diff
@@ -10,184 +9,131 @@
-        return files_list.sort_values(by="wav_filesize")        \
-                         .ix[:, ["wav_filename", "transcript"]] \
-                         .values
-
-    def _indices(self):
+    def _populate_batch_queue(self, session, coord):
+        file_count = len(self.data_set.files)
         index = -1
-        while not self._coord.should_stop():
-            index = self._next_index(index) % len(self._files_list)
-            yield self._files_list[index]
-
-    def _populate_batch_queue(self, session):
-        for wav_file, transcript in self._indices():
-            source = audiofile_to_input_vector(wav_file, self._numcep, self._numcontext)
+        while not coord.should_stop():
+            index = self.data_set.next_index(index) % file_count
+            wav_file, transcript = self.data_set.files[index]
+            source = audiofile_to_input_vector(wav_file, self.data_sets.numcep, self.data_sets.numcontext)
             source_len = len(source)
             target = text_to_char_array(transcript)
             target_len = len(target)
             try:
-                session.run(self._enqueue_op, feed_dict={
-                    self._x: source,
-                    self._x_length: source_len,
-                    self._y: target,
-                    self._y_length: target_len})
+                session.run(self._enqueue_op, feed_dict={ self.data_sets.ph_x: source,
+                                                          self.data_sets.ph_x_length: source_len,
+                                                          self.data_sets.ph_y: target,
+                                                          self.data_sets.ph_y_length: target_len })
             except tf.errors.CancelledError:
                 return
 
-    def next_batch(self):
-        source, source_lengths, target, target_lengths = self.example_queue.dequeue_many(self.batch_size)
-        sparse_labels = ctc_label_dense_to_sparse(target, target_lengths, self.batch_size)
-        return source, source_lengths, sparse_labels
-
-    @property
-    def total_batches(self):
-        # Note: If len(_files_list) % batch_size != 0, this re-uses initial files
-        return int(ceil(len(self._files_list) / self.batch_size))
-
-class SwitchableDataSet(object):
-    def __init__(self, data_sets):
-        '''Data set that is wrapping a data sets instance to switch between train, dev and test instances during training.
-
-        Args:
-            data_sets (DataSets): the corpus container holding all three data sets
-        '''
-        self._data_sets = data_sets
-        self._sets = [data_sets.train, data_sets.dev, data_sets.test]
-        self._queues = [s.example_queue for s in self._sets]
-        self._queue_selector = tf.placeholder(tf.int32, name='Queue_Selector')
-        self._queue = tf.QueueBase.from_list(self._queue_selector, self._queues)
+class GpuFeeder(object):
+    def __init__(self, data_sets, index):
+        self.data_sets = data_sets
```

Comment:
PEP-8 recommends a single leading underscore when naming non-public instance variables. I'd rather stay in line with it, so that for example in the future it's easier to add linting.